### PR TITLE
Modernization never ends

### PR DESCRIPTION
**Pull request by gwynne**
- Significant modernizations to the `Dockerfile` and package manifest to improve build behavior.
- No longer relies on Vapor-specific Docker images. We can now use the Swift base images directly!
- Various other minor improvements.

(Note: This PR is marked as a bugfix because it fixes the issue of the `Dockerfile` attempting to deploy the built app to `/run` in the final runner image, which was not a valid or safe location for it.)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,21 +20,17 @@ RUN swift package resolve
 COPY . .
 
 # Compile with optimizations
-RUN swift build \
-	--enable-test-discovery \
-	-c release \
-	-Xswiftc -g
+RUN swift build --enable-test-discovery -c release
```
Comment (0xTim) [inline Dockerfile:23]: @gwynne why did you remove the `-Xswiftc -g` flag? That's required for all the backtrace stuff to work

Comment (gwynne) [reply Dockerfile:23]: It's automatically included even in release builds since Swift 5.2, not sure if it was a point release that made that particular change but I know it's been the case since at least 5.2.2. Add `-v` to show the compiler invocations if you want to verify, I certainly wouldn't mind the confirmation.
